monkestation/code/modules/power/rbmk/fuel_rods.dm

```diff
@@ -0,0 +1,246 @@
+
+/obj/item/fuel_rod/material/attackby(obj/item/W, mob/user, params)
+	var/obj/item/stack/M = W
+	if(istype(M, material_type))
+		if(!check_material_input(user))
+			return
+		if(initial_amount < max_initial_amount)
+			var/adding = min((max_initial_amount - initial_amount), M.amount)
+			M.amount -= adding
+			initial_amount += adding
+			if(adding == 1)
+				to_chat(user, "<span class='notice'>You insert [adding] [material_name_singular] into \the [src].</span>")
+			else
+				to_chat(user, "<span class='notice'>You insert [adding] [material_name] into \the [src].</span>")
+			M.zero_amount()
+		else
+			to_chat(user, "<span class='warning'>\The [src]'s material slots are full!</span>")
+			return
+	else
+		return ..()
+
+/obj/item/fuel_rod/material/attack_self(mob/user)
+	if(expended)
+		to_chat(user, "<span class='notice'>You have already removed [material_name] from \the [src].</span>")
+		return
+
+	if(depleted_final)
+		new material_type(user.loc, grown_amount)
+		if(grown_amount == 1)
+			to_chat(user, "<span class='notice'>You harvest [grown_amount] [material_name_singular] from \the [src].</span>") // Unlikely
+		else
+			to_chat(user, "<span class='notice'>You harvest [grown_amount] [material_name] from \the [src].</span>")
+		playsound(loc, 'sound/effects/stonedoor_openclose.ogg', 50, 1)
+		grown_amount = 0
+		expend()
+		return
+	if(depletion)
+		to_chat(user, "<span class='warning'>\The [src] has not fissiled enough to fully grow the sample. The progress bar shows it is [min(depletion/depletion_threshold*100,100)]% complete.</span>")
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, 0)
+		return
+	if(initial_amount)
+		new material_type(user.loc, initial_amount)
+		if(initial_amount == 1)
+			to_chat(user, "<span class='notice'>You remove [initial_amount] [material_name_singular] from \the [src].</span>")
+		else
+			to_chat(user, "<span class='notice'>You remove [initial_amount] [material_name] from \the [src].</span>")
+		playsound(loc, 'sound/effects/stonedoor_openclose.ogg', 50, 1)
+		initial_amount = 0
+
+/obj/item/fuel_rod/material/examine(mob/user)
+	. = ..()
+	if(expended)
+		. += "<span class='warning'>The material slots have been slagged by the extreme heat, you can't grow [material_name] in this rod again...</span>"
+		return
+	else if(depleted_final)
+		if(grown_amount == 1)
+			. += "<span class='warning'>This fuel rod's [material_name] are now fully grown, and it currently bears [grown_amount] harvestable [material_name_singular].</span>"
+		else
+			. += "<span class='warning'>This fuel rod's [material_name] are now fully grown, and it currently bears [grown_amount] harvestable [material_name].</span>"
+		return
+	if(depletion)
+		. += "<span class='danger'>The sample is [min(depletion/depletion_threshold*100,100)]% fissiled.</span>"
+	. += "<span class='disarm'>[initial_amount]/[max_initial_amount] of the slots for [material_name] are full.</span>"
+
+/obj/item/fuel_rod/material/telecrystal
```

Comment:
Absolutely no telecrystal growth

Reply:
It is only available if it is added to the traitor menu